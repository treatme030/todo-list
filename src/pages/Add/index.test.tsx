import React from "react";
import { Router, useLocation } from "react-router-dom";
import { createMemoryHistory } from 'history';
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import 'jest-styled-components';
import { ToDoListProvider } from "contexts";
import { Add } from "./index";

describe('<Add />', () => {
  it('renders component correctly', () => {
    const history = createMemoryHistory();
    history.push('/add');

    const { container } = render(
      <Router history={history}>
        <Add />
      </Router>
    );

    const input = screen.getByPlaceholderText('할 일을 입력해 주세요');
    expect(input).toBeInTheDocument();
    const addButton = screen.getByText('추가');
    expect(addButton).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
  it('add a new ToDo and redirect to the root page', () => {
    const history = createMemoryHistory();
    history.push('/add');

    const TestComponent = () => {
      const { pathname } = useLocation();
      return (
        <ToDoListProvider>
          <div>{pathname}</div>
          <Add />
        </ToDoListProvider>
      );
    };

    render(
      <Router history={history}>
        <TestComponent />
      </Router>
    );

    const pathName = screen.getByText('/add');
    expect(pathName).toBeInTheDocument();

    const input = screen.getByPlaceholderText('할 일을 입력해 주세요');
    const addButton = screen.getByText('추가');
    userEvent.type(input, 'New ToDo');
    userEvent.click(addButton);
    expect(pathName.textContent).toBe('/');
    expect(JSON.parse(localStorage.getItem('ToDoList') as string)).toContain('New ToDo');
  });
});