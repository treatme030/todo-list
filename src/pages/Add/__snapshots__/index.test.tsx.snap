// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Add /> renders component correctly 1`] = `
.c3 {
  text-align: center;
  background-color: #304ffe;
  padding: 10px 20px;
  border-radius: 8px;
  cursor: pointer;
}

.c3:hover {
  background-color: #1e40fe;
}

.c3:active {
  box-shadow: inset 5px 5px 10px rgba(0,0,0,0.2);
}

.c4 {
  color: #fff;
  font-size: 16px;
}

.c2 {
  font-size: 16px;
  padding: 10px 10px;
  border-radius: 8px;
  border: 1px solid #bdbdbd;
  outline: none;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #fff;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 20px;
  border-radius: 8px;
  box-shadow: 5px 5px 10px rgba(0,0,0,0,2);
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <input
        class="c2"
        placeholder="할 일을 입력해 주세요"
        value=""
      />
      <button
        class="c3"
      >
        <label
          class="c4"
        >
          추가
        </label>
      </button>
    </div>
  </div>
</div>
`;
